lib/Controller/VideoController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * @copyright Copyright (c) 2021 Frederic Ruget <fred@atlant.is>
+ *
+ * @author Frederic Ruget (douzebis) <fred@atlant.is>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Viewer\Controller;
+
+use Exception;
+
+use OCP\IRequest;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\Controller;
+use OCP\L10N\IFactory;
+
+use OCP\Files\File;
+use OCP\Files\IRootFolder;
+use OCP\IUserSession;
+
+class VideoController extends Controller {
+
+	// I copied the annotation below without understanding it... is it useful?
+	/** @var IRootFolder */
+	private $storage;
+
+	// I copied the annotation below without understanding it... is it useful?
+	/** @var array */
+	private $locales;
+
+	public function __construct(
+		string $AppName,
+		IRequest $request,
+		IRootFolder $storage,
+		IFactory $factory,
+		IUserSession $userSession
+	){
+		parent::__construct($AppName, $request);
+		$this->storage = $storage;
+		$this->locales = $factory->findAvailableLocales();;
+		$this->currentUser = $userSession->getUser();
+	}
+
+	// I copied the annotation below without understanding it... is it useful?
+	/**
+	 * @NoAdminRequired
+	 */
+	public function getTracks($videoPath): DataResponse  {
+		// $videoPath is the path to the main video file, eg: 'mypath/movie.mkv'
+		// Return an array of found tracks associated with the main video.
+		// Each track is an object:
+		//   - basename  /* basename of the track */
+		//   - language  /* language for the track */
+		//   - locale    /* (usuallu) 2-letter code for the language */
+		//
+		$video = $this->storage->getUserFolder($this->currentUser->getUID())
+			->get($videoPath);
+		$rootName = pathinfo($video->getFileInfo()->getPath())['filename'];
+		$rootLen = strlen($rootName);
+		// eg: $rootName is 'movie', $rooLen is 5
+		$files = $video->getParent()->getDirectoryListing();
```

Comment:
What does this line return when only the video file is shared? (And not its folder)

Reply:
Good question.
I made a test with the following setup: Alice shares 'movie.mkv' with John, but does not share 'movie.en.vtt'.
When john open the video for viewing, 'getDirectoryListing()' returns a shorter list of files, including 'movie.mkv', but not 'movie.en.vtt'. As a result, John does not have access to captions.

Does this answer the question? or maybe you had a different setup in mind?

Reply:
I had the public share setup in mind. I don't know what `getParent` returns if the video would be shared with a link

Reply:
Good call for the public side of this feature

Reply:
@beardhatcode I did a second test today.

Sharing 'movie.mkv' (and only this file) using federation sharing -- john@https://sandbox.nc.com (remote)
There was no problem for john viewing the movie (without captions of course).
Single stepping the php controller,`getParent()` seems to return the "local" parent folder, not the "remote" parent folder where the file is actually hosted.

Btw, is there a specification of the expected behavior for `getParent()`, including exception cases and/or error return values?

Reply:
See the code: https://github.com/nextcloud/server/blob/master/lib/public/Files/Node.php#L234 you can click on getParrent to find all the implementations of it for various types of files

The "normal" one is here: https://github.com/nextcloud/server/blob/f2101487a0a32cd5be5068329a2c7cd6d8a1583f/lib/private/Files/Node/Node.php#L282-L288  , it appears that the root folder is returned whenever the dirname cannot be obtained. I can't figure out what happens when te file is shared, in the share20 folder "setParent" is marked as deprecated ... (@skjnldsv do you know how this works?)

Reply:
OK. I think it is OK then.
The path is always from the point of view of the local user file space.
Regarding the "normal" implementation, I understand it maps "" (empty path), ".",  and "/" to the root directory of the local user. (It does not necessarily mean that the dirname could not be obtained.)

Reply:
@beardhatcode I believe I took most of the review comments into account.
Is there more to be done about the `getParent` issue?